### PR TITLE
Merge 17.6 beta 1 code freeze into develop

### PR DESCRIPTION
**Pull request by mokagio**
Merges the 17.6 beta 1 code freeze changes into `develop`.

To test, review the code. If CI is green, we're good.

## Regression Notes
1. Potential unintended areas of impact
N.A.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N.A.

3. What automated tests I added (or what prevented me from doing so)
N.A.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes. – N.A.
- [x] I have considered adding accessibility improvements for my changes. –  N.A.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.  – N.A.

### CODE REVIEW

**Review by mokagio — COMMENTED** [review]
- `Podfile` changes as a result of publishing stable versions of the internal pods to start the new release cycle
- New `RELEASE-NOTES.txt` header for 17.7
- `release_notes.txt` is a copy of the 17.6 `RELEASE-NOTES.txt` content as per changes in #16674, this way the beta release on GitHub has the correct notes
- Version bumps in the `.xcconfig` files
- Version update in the Fastlane-based automation files
- No `.po[t]` changes because the release notes haven't been run through editorial yet
- I'm not sure about that negative diff in the `en.lproj` file, but the file is still there 😅 I suppose some strings have been removed in the dependencies and that's the result of the localization automation?

File: WordPress/Resources/release_notes.txt [hunk 1]
```diff
@@ -1,17 +1,11 @@
-What’s new in WordPress release 17.5?
+* [**] Reader Post details: now shows a summary of Likes for the post. Tapping it displays the full list of Likes. [#16628]
```
Comment (jkmassel) [inline WordPress/Resources/release_notes.txt:1]: Ideally I think we'd get rid of the priority indicators and PR numbers in this file prior to landing it, but I don't recall if this is a tooling-generated change.

No need to fix it for this PR, but if it was by hand, it'd be nice for next time – just makes the release notes a bit nicer in GitHub and such :)

File: fastlane/Fastfile [hunk 2]
```diff
@@ -34,7 +34,7 @@ ALL_BUNDLE_IDENTIFIERS = [
   'org.wordpress.WordPressAllTimeWidget',
   'org.wordpress.WordPressThisWeekWidget',
   'org.wordpress.WordPressIntents'
-].freeze
+]
```
Comment (jkmassel) [inline fastlane/Fastfile:37]: Why don't we want to `.freeze` this? I can't imagine we want it mutable? (not blocking)

**Review by jkmassel — APPROVED** [review]
A couple nits, but nothing blocking – let's land it and address them (if needed) later

File: WordPress/Resources/release_notes.txt (hunk 1 above)
Comment (mokagio) [reply WordPress/Resources/release_notes.txt:1]: Gotcha! I'll keep it in mind 👌

File: fastlane/Fastfile (hunk 2 above)
Comment (mokagio) [reply fastlane/Fastfile:37]: We don't, but `match` does 😭 

Under the hood, it calls `.flatten!` on the value which [crashes](https://app.circleci.com/pipelines/github/wordpress-mobile/WordPress-iOS/22013/workflows/b25664b2-3ce6-49f4-a795-818a9673f67e/jobs/50408) if this is a constant.

See also:

- fastlane/fastlane#11324
- fastlane/fastlane@b817148/match/lib/match/runner.rb#L84-L86

**Pull request merged**